estate/models/estate_property.py

```diff
@@ -0,0 +1,105 @@
+from dateutil.relativedelta import relativedelta
+from odoo import models, api, fields, exceptions
+from odoo.exceptions import ValidationError
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+    _check_expected_price = models.Constraint(
+        "CHECK(expected_price > 0)", "Expected price must be strictly positive."
+    )
+    _check_selling_price = models.Constraint(
+        "CHECK(selling_price >= 0)", "Selling price must be positive."
+    )
```

Comment:
It is good if we can declare it after the field declaration. You can refer odoo codebase.